nixos/modules/programs/qgroundcontrol.nix

```diff
@@ -0,0 +1,55 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.qgroundcontrol;
+in
+{
+
+  options = {
+    programs.qgroundcontrol = {
+      enable = lib.mkEnableOption "qgroundcontrol";
+
+      package = lib.mkPackageOption pkgs "qgroundcontrol" {};
+
+      udevRules = {
+        enable = lib.mkEnableOption "udev rules for qgroundcontrol" // {
+          default = true;
+        };
+
+        rules = lib.mkOption {
+          description = ''
+            udev rules for qgroundcontrol to access serial ports.
+
+            These are quite permissive and may not be suitable for systems with untrusted users.
+            In that case, you may want to adjust these rules to only allow access to specific devices.
+          '';
+          type = lib.types.string;
+          default = ''
+            SUBSYSTEM=="tty", MODE="0666"
+          '';
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    # ModemManager is known to interfere with serial connections;
+    # qgroundcontrol recommends disabling it
+    systemd.services.ModemManager.enable = lib.mkDefault false;
+
+    # qgroundcontrol requires access to serial ports to communicate with drones
+    udev.extraRules = lib.optionalString cfg.udevRules.enable ''
+      # nixos/qgroundcontrol
+      ${cfg.udevRules.rules}
+    '';
+  };
+
+  meta.maintainers = pkgs.qgroundcontrol.meta.maintainers ++ (with lib.maintainers; [ pandapip1 ]);
```

Comment:
Would you mind pinging the maintainer(s) of the package unless you've already done so? Maintaining a module is quite different from a package, and I don't think I would like it if I were just silently marked as the maintainer of something I wasn't given any notice about.